data_structures/src/chain/priority.rs

```diff
@@ -0,0 +1,916 @@
+║      Low │ {:<16} │ {:<25}  ║
+║   Medium │ {:<16} │ {:<25}  ║
+║     High │ {:<16} │ {:<25}  ║
+║  Opulent │ {:<16} │ {:<25}  ║
+╚══════════════════════════════════════════════════════════╝"#,
+            // Believe it or not, these `to_string` are needed for proper formatting, hence the
+            // clippy allow directive above.
+            self.drt_stinky.priority.to_string(),
+            self.drt_stinky.time_to_block.to_string(),
+            self.drt_low.priority.to_string(),
+            self.drt_low.time_to_block.to_string(),
+            self.drt_medium.priority.to_string(),
+            self.drt_medium.time_to_block.to_string(),
+            self.drt_high.priority.to_string(),
+            self.drt_high.time_to_block.to_string(),
+            self.drt_opulent.priority.to_string(),
+            self.drt_opulent.time_to_block.to_string(),
+            self.vtt_stinky.priority.to_string(),
+            self.vtt_stinky.time_to_block.to_string(),
+            self.vtt_low.priority.to_string(),
+            self.vtt_low.time_to_block.to_string(),
+            self.vtt_medium.priority.to_string(),
+            self.vtt_medium.time_to_block.to_string(),
+            self.vtt_high.priority.to_string(),
+            self.vtt_high.time_to_block.to_string(),
+            self.vtt_opulent.priority.to_string(),
+            self.vtt_opulent.time_to_block.to_string(),
+        )
+    }
+}
+
+/// A estimate for priority and time-to-block.
+///
+/// Time-to-block states what is the expected time (in epochs) that it would take for a transaction
+/// with this priority to be included into a block.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct PriorityEstimate {
+    pub priority: Priority,
+    pub time_to_block: TimeToBlock,
+}
+
+impl fmt::Display for PriorityEstimate {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{:<16} | {}", self.priority, self.time_to_block)
+    }
+}
+
+/// Allows tagging time-to-block estimations for the sake of UX.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub enum TimeToBlock {
+    /// The time-to-block is around X epochs.
+    Around(usize),
+    /// The time-to-block is less than X epochs.
+    LessThan(usize),
+    /// The time-to-block is unknown.
+    #[default]
+    Unknown,
+    /// The time-to-block is up to X epochs.
+    UpTo(usize),
+}
+
+impl fmt::Display for TimeToBlock {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            TimeToBlock::Around(x) => write!(f, "around {} epochs", x),
+            TimeToBlock::LessThan(x) => write!(f, "less than {} epochs", x),
+            TimeToBlock::Unknown => write!(f, "unknown"),
+            TimeToBlock::UpTo(x) => write!(f, "up to {} epochs", x),
+        }
+    }
+}
+
+/// Calculates the average gap between the values in a `Vec<usize>`.
+fn average_gap(occurrences: Vec<usize>, sample_size: usize) -> usize {
```

Comment:
I expected this function to return a float, in order to be able to provide a better average. Also some tests to ensure that it works properly would be nice.